torch/csrc/jit/pybind_utils.h

```diff
@@ -171,6 +235,18 @@ inline InferredType tryToInferContainerType(py::handle input) {
         return type_match.reason();
       }
     }
+    py::object input_type =py::cast<py::object>(input.get_type());
+    if (isNamedTupleClass(input_type)) {
+      // If the tuple is a NamedTuple type
+      auto qualifiedName = c10::QualifiedName(py::cast<std::string>(
+          py::module::import("torch.jit").attr("_qualified_name")(input_type)));
+      c10::TypePtr named_type = getOrCreateNamedTupleType(
+          input_type,
+          qualifiedName,
+          element_types,
```

Comment:
what happens if the same named tuple is used with different types into two different places in the trace?

Reply:
same namedtuple should have a single named tuple type in the python cu because we drop the second one if the first one is a subtype of the second one.